### ISSUE TITLE
tar: Error opening archive: Failed to open 'tensorlow_macos-0.1alpha2.tar'

### ISSUE THREAD

**Issue by NdodaEnde** [apple/tensorflow_macos#162]
I am struggling to to extract the tar file. this is the error I get. kindly assist.

**Comment by pooyadavoodi** [comment]
Thanks for trying out the new release. 

From the title of the issue, it seems that the filename you are using is missing `.gz` at the end. 

Could you please verify that you are using the right filename and verify that you have the file in your directory.

The name of the tarball that we have uploaded is this: `tensorflow_macos-0.1alpha2.tar.gz`.

**Comment by NdodaEnde** [comment]
Hi Pooya,
Thanks, I managed to fix the problem. Now I am stuck here:



Kindly assist.

**Comment by NdodaEnde** [comment]
Here’s my screenshot for contents inside arm64:

**Comment by anna-tikhonova** [comment]
@NdodaEnde I am not able to see the attachments in your comments. Could you try to attach them again?

**Comment by NdodaEnde** [comment]
Hi…I am on the last step, installing tensor board.this is the error I get



Also, the TensorFlow_addons file in my downloaded files doesn’t have the mlcompute…am not sure what significance of it.

**Comment by anna-tikhonova** [comment]
@NdodaEnde I only see text in your comments; I do not see any attachments or images. Are you attempting to post screenshots? Perhaps, instead of posting screenshots, you could copy&paste the errors you are seeing as text?

**Comment by NdodaEnde** [comment]
Okay,
This is the error I get when trying to pip install tensor board:
ERROR: failed building wheel for grpcio
Running setup.py clean for grpcio
Failed to build grpcio

I then deactivated the environment and created a new one, then I started the whole installation process from scratch….same thing when it comes to installing tensorboard.
I am at a wits end, I don’t know what else to try.

**Comment by pooyadavoodi** [comment]
Could you share how you install the package?
I suggest to install grpcio from the package (grpcio-1.33.2-cp38-cp38-macosx_11_0_arm64.whl), and then install tensorboard which should then be satisfied with the installed grpcio.

In general, I recommend to use `install_venv.sh` which comes with the package.

**Comment by NdodaEnde** [comment]
Thank your for your prompt response. I used this to install:
pip install –-upgrade -t “$env/lib/python3.8/site-packages/” –-no-dependencies –force $libs/grpcio-1.33.2-cp38-cp38-macosx_11_0_arm64.whl

So do I go and install venv.sh…like you suggested below and then pip install tensorboard?

**Comment by pooyadavoodi** [comment]
Sure, using `./install_venv.sh -p` would be the best. This also installs tensorboard: https://github.com/apple/tensorflow_macos/blob/master/scripts/install_venv.sh#L335

Before running `install_venv.sh`, I suggest to deactivate your current venv; and either delete the venv or make sure you use a different name for the new venv that `install_venv.sh` creates.

**Comment by NdodaEnde** [comment]
I’ll just deactivate the virtual environment and start from afresh. Is the link you sent below have all the steps of installing?

**Comment by pooyadavoodi** [comment]
Yes, you would only need to download the tarball, extract it, and then run `install_venv.sh`. Then TF should be ready to use.

**Comment by NdodaEnde** [comment]
Cool…thanks, will give it a shot

**Comment by NdodaEnde** [comment]
So below is the steps I have been following before I got stuck on tensorboard. As you can see I was stuck on the last point.
So I wanted to find out from you, after which step do I follow your instructions i.e. installing venv.sh?


Install miniforge3: https://github.com/conda-forge/miniforge/#download <https://github.com/conda-forge/miniforge/#download>
Apple Silicon link
Restart terminal so that Conda is recognized
Type Conda in prompt to check
Create Conda environment with python3.8 since python3.9 is the default from Miniforge (user python38 for your name): conda create —name python38-demo-v2 python=3.8
Activate virtual environment: conda activate <env_name>
Download the zipped tar file from apple/tensorflow_macos
https://github.com/apple/tensorflow_macos/releases <https://github.com/apple/tensorflow_macos/releases>
Unzip the tar file
Navigate to Downloads
Tar -xvf <filename>
Get the working directory of arm64 within that folder
Once there, type: pwd
Set variables
libs=location of arm64 above
libs=“/Users/nick/Downloads/tensorflow_macos/arm64/“
env=location of your python virtual environment
env=“/Users/nick/miniforge3/envs/<env_name>
Conda install cached-property
Conda install six
pip install –upgrade -t “$env/lib/python3.8/site-packages/” –no-dependencies –force “$libs/grpcio-1.33.2-cp38-cp38-macosx_11_0_arm64.whl”
pip install –upgrade -t “$env/lib/python3.8/site-packages/” –no-dependencies –force “$libs/h5py-2.10.0-cp38-cp38-macosx_11_0_arm64.whl”
pip install –upgrade -t “$env/lib/python3.8/site-packages/” –no-dependencies –force “$libs/tensorflow_addons-0.11.2+mlcompute-cp38-cp38-macosx_11_0_arm64.whl”
conda install -c conda-forge -y absl-py
conda install -c conda-forge -y astunparse

conda install -c conda-forge -y gast

conda install -c conda-forge -y opt_einsum

conda install -c conda-forge -y termcolor

conda install -c conda-forge -y typing_extensions

conda install -c conda-forge -y wheel

conda install -c conda-forge -y typeguard

14. pip install wrapt flatbuffers tensorflow_estimator google_pasta keras_preprocessing protobuf

15. pip install –upgrade -t “$env/lib/python3.8/site-packages/” –no-dependencies –force “$libs/tensorflow_macos-0.1a1-cp38-cp38-macosx_11_0_arm64.whl”

Make sure this is pointing to the right version within your downloads folder

16. pip install tensorboard

17. type python into the command prompt to open a python shell

18. import tensorflow

**Comment by NdodaEnde** [comment]
So I created a new environment and I then followed these instructions, and it worked.
https://towardsdatascience.com/tensorflow-2-4-on-apple-silicon-m1-installation-under-conda-environment-ba6de962b3b8

**Comment by NdodaEnde** [comment]
Hi Pooya,

Been a great journey so far with my MacBook and deep learning.
A quick one:
I want to install cv2 but the virtual environment I installed tensor flow macos uses an older version of pip. For cv2 installation it requires the latest version. How do I go about it in resolving this conflict?
thanks.
